modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -2948,6 +2948,113 @@ definition {
 			locator1 = "ObjectAdmin#VIEW_ACTIONS_LABEL");
 	}
 
+	@description = "LPS-165849 - Verify terms from a related custom object (child) are replaced correctly in an email notification"
+	@priority = 5
+	test CanSendEmailNotificationWithTermsComingFromCustomObjectRelationship {
+		property custom.properties = "feature.flag.LPS-165849=true";
+		property custom.properties = "feature.flag.LPS-171625=true";
+		property portal.acceptance = "true";
+		property test.smtp.server.enabled = "true";
+
+		task ("Given custom object A with a 1:M relationship to custom object B") {
+			for (var letters : list "A,B") {
+				ObjectAdmin.addObjectViaAPI(
+					labelName = "Custom Object ${letters} 182605",
+					objectName = "CustomObject${letters}182605",
+					pluralLabelName = "Custom Objects ${letters} 182605");
+
+				ObjectAdmin.addObjectFieldViaAPI(
+					fieldBusinessType = "Text",
+					fieldLabelName = "Field Text ${letters}",
+					fieldName = "customField",
+					fieldType = "String",
+					isRequired = "false",
+					objectName = "CustomObject${letters}182605");
+
+				ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject${letters}182605");
+			}
+
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObjectA182605",
+				objectName_2 = "CustomObjectB182605",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a email notification template using terms from both object definitions") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Notifications",
+				panel = "Control Panel",
+				portlet = "Templates");
+
+			ObjectNotifications.addNewNotificationTemplate(
+				emailBody = "[%CUSTOMOBJECTB182605_CUSTOMFIELD%]",
+				entryTo = "[%CUSTOMOBJECTB182605_AUTHOR_EMAIL_ADDRESS%]",
+				fromAddress = "[%RELATIONSHIP_CUSTOMOBJECTA182605_AUTHOR_EMAIL_ADDRESS%]",
+				fromName = "[%CUSTOMOBJECTB182605_AUTHOR_FIRST_NAME%]",
+				name = "Notification Template",
+				notificationType = "Email",
+				subject = "[%RELATIONSHIP_CUSTOMOBJECTA182605_AUTHOR_FIRST_NAME%] [%RELATIONSHIP_CUSTOMOBJECTA182605_AUTHOR_LAST_NAME%]");
+
+			PortletEntry.save();
+		}
+
+		task ("And given the title field of custom object A is changed") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Custom Object A 182605");
+
+			CreateObject.selectTitleField(fieldLabel = "Field Text A");
+
+			CreateObject.saveObject();
+		}
+
+		task ("And given an action in Object 2 to send an email notification after an entry is updated") {
```

Comment:
Since you're using the pattern "A,B,C..." instead "1,2,3...", change here to Object B. "And given an action On After Update to send a email notification created on Object B"